### ISSUE TITLE
Ingestion of invalid attributes dropped without warning

### ISSUE THREAD

**Issue by emuradov** [MISP/PyMISP#855]
When using the WebUI and creating an attribute under an event, selecting a type and value that are not compatible will present an error message with an explanation and deny submission of the attribute.

In comparison, when using PyMISP, invalid attributes can be added to the misp object in memory, but will be dropped without any notice once the request to update an event or a similar function is made.

Is there a way that a PyMISP client can be informed of these validation errors so it knows which attributes did not get ingested into the event?

Still relatively new here and would appreciate the help!

**Comment by github-germ** [comment]
I agree this would be a nice improvement.

**Comment by tomking2** [comment]
I'll try to take a look into what information is returned during ingestion of an event in the coming days to see if MISP provides any context around what was dropped. I assume it doesn't return this information (as it directly returns the event) but we could probably implement a validate parameter that performs a simple comparison between in/out. That, or perform stricter validation on adding an attribute to an event in memory

**Comment by Rafiot** [comment]
Yeah, there is a whole bunch of massaging that happens on MISP side when you submit stuff, so just comparing in and out attribute values will match thing that are in fact added, and miss other. And then there is the issue of figuring out why it wasn't added: was it a duplicate? Was it an invalid attribute? And if you push a complete event with updates, you need to iterate over *everything* (attributes, comments, objects, tags...).

I think there is an issue in the MISP repository but I cannot find it anymore...

**Comment by tomking2** [comment]
I guess it does start to get complex when you want to provide the _why_, not just the _what_. I expect a high-level "this stuff wasn't added" shouldn't be too difficult, it's the "why wasn't it added" will cause some head scratching.

I'm tempted to suggest this might be more a MISP problem than a PyMISP problem, as it's best placed to report on submission errors. if you hunt down the issue, let me know

**Comment by Rafiot** [comment]
This issue is talking about that: https://github.com/MISP/MISP/issues/5257

But yes, as you said, we can probably give some feedback to the user when they add attributes, but for anything else, it's not really an option.

**Comment by emuradov** [comment]
The issue that I have experienced was specifically invalid attributes being dropped, but I do believe duplicates fall into the same category. An example would be including an invalid IP port value such as 0. It does seem like the problem extends beyond PyMISP as validation of attributes values does not seem to occur
